### PR TITLE
'Print Manifest' fix

### PR DESCRIPTION
**Pull request by PatrickWaweru**
1. Enable 'Print Manifest' button only after the full manifest is submitted
2. Print manifest order ordered by date and time sent ascending
3. fix sample type bug - select field should only have valid values 
4. make sure an order can only be added once per manifest using a DB unique index

### CODE REVIEW

File: omod/src/main/webapp/pages/orders/manifestOrdersHome.gsp [hunk 1]
```diff
@@ -98,7 +98,7 @@ tr:nth-child(even) {background-color: #f2f2f2;}
             </button>
                     </td>
             <% } %>
-            <% if (manifestOrders.size() > 0) { %>
+            <% if (manifestOrders.size() > 0 && manifest.status.trim().toLowerCase() == 'submitted') { %>
```
Comment (ojwanganto) [inline omod/src/main/webapp/pages/orders/manifestOrdersHome.gsp:101]: we can check for non-null on status just so we avoid NPE

File: api/src/main/java/org/openmrs/module/kenyaemrorderentry/labDataExchange/LabwareSystemWebRequest.java [hunk 2]
```diff
@@ -157,6 +157,7 @@ public boolean postSamples(LabManifestOrder manifestOrder, String manifestStatus
                 return(false);
             } else if (statusCode == 201 || statusCode == 200) {
                 manifestOrder.setStatus("Sent");
+                manifestOrder.setDateSent(new Date());
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemrorderentry/labDataExchange/LabwareSystemWebRequest.java:160]: We can add this to CHAI as well. We need the timestamp

Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemrorderentry/labDataExchange/LabwareSystemWebRequest.java:160]: We can add this to CHAI as well. We need the timestamp

**Review by ojwanganto — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/kenyaemrorderentry/labDataExchange/LabwareSystemWebRequest.java (hunk 2 above)
Comment (PatrickWaweru) [reply api/src/main/java/org/openmrs/module/kenyaemrorderentry/labDataExchange/LabwareSystemWebRequest.java:160]: This was already added on the CHAI system

File: omod/src/main/webapp/pages/orders/manifestOrdersHome.gsp (hunk 1 above)
Comment (PatrickWaweru) [reply omod/src/main/webapp/pages/orders/manifestOrdersHome.gsp:101]: Ok

Comment (PatrickWaweru) [reply omod/src/main/webapp/pages/orders/manifestOrdersHome.gsp]: Fixed here: https://github.com/palladiumkenya/openmrs-module-kenyaemrorderentry/pull/247/commits/d038433c3adc553031ba468181f647aff0b14af7

File: api/src/main/java/org/openmrs/module/kenyaemrorderentry/labDataExchange/LabwareSystemWebRequest.java (hunk 2 above)
Comment (PatrickWaweru) [reply api/src/main/java/org/openmrs/module/kenyaemrorderentry/labDataExchange/LabwareSystemWebRequest.java:160]: This was already added on the CHAI system

**Review by ojwanganto — APPROVED** [review]

**Pull request merged**